rest-api/src/main/java/com/epam/esm/exceptionhandler/RestExceptionHandler.java

```diff
@@ -116,11 +116,58 @@
 	public ResponseEntity<ErrorData> constraintExceptionHandler(ConstraintViolationException exception) {
 		log.error(exception.getLocalizedMessage(), exception);
 		String message = exception.getLocalizedMessage();
-		String code = HttpStatus.BAD_REQUEST.value() + ErrorCode.INCORRECT_PARAM.getErrorCode();
+		String code = HttpStatus.BAD_REQUEST.value() + ErrorCode.INCORRECT_PARAM;
 		ErrorData incorrectData = new ErrorData(List.of(message), code);
 		return new ResponseEntity<>(incorrectData, HttpStatus.BAD_REQUEST);
 	}
 	
+	
+	
+	
+	
+	
+    /**
+     * Handles BadCredentialsException
+     *
+     * @param exception {@link BadCredentialsException} exception
+	 * @param locale {@link Locale} locale of HTTP request
+	 * @return {@link ResponseEntity} the response message
+     */
+    @ExceptionHandler(BadCredentialsException.class)
+    @ResponseStatus(HttpStatus.UNAUTHORIZED)
+    public ResponseEntity<ErrorData> badCredentialsExceptionHandler(BadCredentialsException exception, Locale locale) {
+    	log.error(exception.getLocalizedMessage(), exception);
+    	String message = messageSource.getMessage(ErrorMessageKey.BAD_CREDENTIALS, new String[]{}, locale);
+		String code = HttpStatus.UNAUTHORIZED.value() + ErrorCode.UNAUTHORIZED;
```

Comment:
please fix tab/spaces, you are using mixed tabs with spaces, and because of this your code unreadable in git

Reply:
in setting tab=4, fixed tabs and spaces